File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyInvitationDungeonDetailsMessage.scala





















// Generated on 05/03/2013 10:34:47
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyInvitationDungeonDetailsMessage extends PartyInvitationDetailsMessage {
  override val messageId = 6262

  protected trait Message extends PartyInvitationDetailsMessage.Message with Serializable {
    override val messageId = PartyInvitationDungeonDetailsMessage.messageId

    val dungeonId: Short
    val playersDungeonReady: Seq[Boolean]

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(dungeonId)
      buf.writeUShort(playersDungeonReady.length)
      for (entry <- playersDungeonReady) {
        buf.writeBoolean(entry)
      }
    }

  }

  protected def create(partyId: Int, partyType: Byte, fromId: Int, fromName: String, leaderId: Int, members: Seq[PartyInvitationMemberInformations.TypeImpl], guests: Seq[PartyGuestInformations.TypeImpl], dungeonId: Short, playersDungeonReady: Seq[Boolean]): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val partyType = buf.readByte()
    val fromId = buf.readInt()
    val fromName = buf.readString()
    val leaderId = buf.readInt()
    val members = {
      val builder = collection.immutable.Seq.newBuilder[PartyInvitationMemberInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += PartyInvitationMemberInformations.deserialize(buf)
      }
      builder.result()
    }
    val guests = {
      val builder = collection.immutable.Seq.newBuilder[PartyGuestInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += PartyGuestInformations.deserialize(buf)
      }
      builder.result()
    }
    val dungeonId = buf.readShort()
    val playersDungeonReady = {
      val builder = collection.immutable.Seq.newBuilder[Boolean]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readBoolean()
      }
      builder.result()
    }

    create(partyId, partyType, fromId, fromName, leaderId, members, guests, dungeonId, playersDungeonReady)
  }

}

object PartyInvitationDungeonDetailsMessage extends PartyInvitationDungeonDetailsMessage {
  case class MessageImpl(partyId: Int, partyType: Byte, fromId: Int, fromName: String, leaderId: Int, members: Seq[PartyInvitationMemberInformations.TypeImpl], guests: Seq[PartyGuestInformations.TypeImpl], dungeonId: Short, playersDungeonReady: Seq[Boolean]) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, partyType: Byte, fromId: Int, fromName: String, leaderId: Int, members: Seq[PartyInvitationMemberInformations.TypeImpl], guests: Seq[PartyGuestInformations.TypeImpl], dungeonId: Short, playersDungeonReady: Seq[Boolean]) = MessageImpl(partyId, partyType, fromId, fromName, leaderId, members, guests, dungeonId, playersDungeonReady)

  def apply(partyId: Int, partyType: Byte, fromId: Int, fromName: String, leaderId: Int, members: Seq[PartyInvitationMemberInformations.TypeImpl], guests: Seq[PartyGuestInformations.TypeImpl], dungeonId: Short, playersDungeonReady: Seq[Boolean]) = create(partyId, partyType, fromId, fromName, leaderId, members, guests, dungeonId, playersDungeonReady)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PartyInvitationDungeonDetailsMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyInvitationDungeonDetailsMessage extends PartyInvitationDetailsMessage {
    public static final int MESSAGE_ID = 6262;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short dungeonId;
    public boolean[] playersDungeonReady;
    
    public PartyInvitationDungeonDetailsMessage() { }
    
    public PartyInvitationDungeonDetailsMessage(int partyId, byte partyType, int fromId, String fromName, int leaderId, PartyInvitationMemberInformations[] members, PartyGuestInformations[] guests, short dungeonId, boolean[] playersDungeonReady) {
        super(partyId, partyType, fromId, fromName, leaderId, members, guests);
        this.dungeonId = dungeonId;
        this.playersDungeonReady = playersDungeonReady;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(dungeonId);
        buf.writeUShort(playersDungeonReady.length);
        for (boolean entry : playersDungeonReady) {
            buf.writeBoolean(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        dungeonId = buf.readShort();
        if (dungeonId < 0)
            throw new RuntimeException("Forbidden value on dungeonId = " + dungeonId + ", it doesn't respect the following condition : dungeonId < 0");
        int limit = buf.readUShort();
        playersDungeonReady = new boolean[limit];
        for (int i = 0; i < limit; i++) {
            playersDungeonReady[i] = buf.readBoolean();
        }
    }
    
}
